### PR TITLE
Add ChunkN to ZStream

### PR DESCRIPTION
**Pull request by LukaszByczynski**
Adds a small helper method to provide user friendly way to chunk the input stream into batches.

Resolves: #1732

### CODE REVIEW

File: streams/shared/src/main/scala/zio/stream/ZStream.scala [hunk 1]
```diff
@@ -664,6 +664,12 @@ class ZStream[-R, +E, +A](val process: ZManaged[R, E, Pull[R, E, A]]) extends Se
     }
   }
 
+  /*** Chunks the stream with specifed chunkSize
+   * @parm chunkSize size of the chunk
+   */
+  def chunkN(chunkSize: Long): ZStream[R, E, List[A]] =
+    transduce(ZSink.collectAllN[A](chunkSize))
```
Comment (iravid) [inline streams/shared/src/main/scala/zio/stream/ZStream.scala]: How about:
```scala
def grouped(size: Long): ZStream[R,  E, List[A]]
def chunkN(size: Long): ZStream[R, E, Chunk[A]]
```

?

Comment (iravid) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala]: `ZStream#grouped` being the same as `List#grouped`

Comment (LukaszByczynski) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala]: Yeah, I did'nt know that we have a Chunk (homework for me).

**Comment by jdegoes** [comment]
@LukaszByczynski @iravid How is this one looking?

File: streams/shared/src/main/scala/zio/stream/ZStream.scala (hunk 1 above)
Comment (LukaszByczynski) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala]: @iravid I have question. How do I should implement this new `ChunkN` version? Is there `ZSink` operator for the Chunk result  ? Maybe, I should imlement one for this case (based on how ZSkink does that for the List)?

Comment (iravid) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala]: Something like `ZSink.collectAllN[A](chunkSize).map(Chunk.fromIterable)`

Comment (iravid) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala]: There's a much more efficient implementation that uses a pre-allocated buffer but we can do that in a follow-up if someone cares about the performance

Comment (LukaszByczynski) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala]: Ok. Tomorrow I will push changes for a review

**Comment by LukaszByczynski** [comment]
@iravid it's ready for a review

File: streams/shared/src/main/scala/zio/stream/ZStream.scala [hunk 2]
```diff
@@ -664,6 +664,12 @@ class ZStream[-R, +E, +A](val process: ZManaged[R, E, Pull[R, E, A]]) extends Se
     }
   }
 
+  /*** Chunks the stream with specifed chunkSize
```
Comment (iravid) [inline streams/shared/src/main/scala/zio/stream/ZStream.scala]:
```suggestion
  /**
   * Chunks the stream with specifed chunkSize
```

File: streams/shared/src/main/scala/zio/stream/ZStream.scala [hunk 3]
```diff
@@ -1268,6 +1274,12 @@ class ZStream[-R, +E, +A](val process: ZManaged[R, E, Pull[R, E, A]]) extends Se
   ): GroupBy[R1, E1, K, A] =
     self.groupBy(a => ZIO.succeed((f(a), a)), buffer)
 
+  /*** Partitions the stream with specifed chunkSize
```
Comment (iravid) [inline streams/shared/src/main/scala/zio/stream/ZStream.scala]:
```suggestion
  /**
   * Partitions the stream with specifed chunkSize
```

**Comment by iravid** [comment]
@LukaszByczynski Could you check the mdoc build failure?

**Comment by LukaszByczynski** [comment]
Fixed

**Comment by iravid** [comment]
Thank you @LukaszByczynski!

**Pull request merged**